node.go

```diff
@@ -1,11 +1,16 @@
 package storm
 
+import "github.com/boltdb/bolt"
+
 // A Node in Storm represents the API to a BoltDB bucket.
 type Node struct {
 	s *DB
 
 	// The root bucket. In the normal, simple case this will be empty.
 	rootBucket []string
+
+	// Transaction object. Nil if not in transaction
+	tx *bolt.Tx
```

Comment:
Hmm... I know this is tempting, but it would be nice if the nodes could be shared. This will break pretty fast with the slightest parallelism ... This will work if you remove all the mutations of Node (n.tx = something) and do a:

```go
func (n Node) WithTransaction(--) *Node
```

Or something.